### PR TITLE
Remove provider from Store WSDL url

### PR DESCRIPTION
**Pull request by tharikaGitHub**
## Purpose
This PR fixes issue https://github.com/wso2/product-apim/issues/4189 for Store WSDL Download url.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java [hunk 1]
```diff
@@ -1170,6 +1170,7 @@ private ConfigParameters() {
     public static final long MAX_FILE_SIZE = 1024L;
 
     public static final String REGISTRY_RESOURCE_PREFIX = "/registry/resource";
+    public static final String REGISTRY_RESOURCE_URL_PREFIX = "/registry/resource/_system/governance/apimgt/applicationdata/provider/";
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java]: line length 120 exceeded?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 2]
```diff
@@ -1740,34 +1751,34 @@ public static ApplicationManagementServiceClient getApplicationManagementService
...
-                    "--" + api.getId().getApiName() + api.getId().getVersion() + ".wsdl";
+            String wsdlResourcePath =
+                    APIConstants.API_WSDL_RESOURCE_LOCATION + createWsdlFileName(api.getId().getProviderName(),
+                            api.getId().getApiName(), api.getId().getVersion());
 
-            String absoluteWSDLResourcePath =
-                    RegistryUtils.getAbsolutePath(RegistryContext.getBaseInstance(),
-                            RegistryConstants.GOVERNANCE_REGISTRY_BASE_PATH) + wsdlResourcePath;
+            String absoluteWSDLResourcePath = RegistryUtils
+                    .getAbsolutePath(RegistryContext.getBaseInstance(), RegistryConstants.GOVERNANCE_REGISTRY_BASE_PATH)
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: line length 120 exceeded?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 3]
```diff
@@ -1740,34 +1751,34 @@ public static ApplicationManagementServiceClient getApplicationManagementService
...
-                        + absoluteWSDLResourcePath;
+                        + RegistryConstants.PATH_SEPARATOR + "resource" + absoluteWSDLResourcePath;
             }
 
             Resource wsdlResource = registry.newResource();
             // isWSDL2Document(api.getWsdlUrl()) method only understands http or file system urls.
             // Hence if this is a registry url, should not go in to the following if block
-            if (!api.getWsdlUrl().matches(wsdRegistryPath) && (api.getWsdlUrl().startsWith("http:") || api.getWsdlUrl
-                    ().startsWith("https:") || api.getWsdlUrl().startsWith("file:"))) {
+            if (!api.getWsdlUrl().matches(wsdRegistryPath) && (api.getWsdlUrl().startsWith("http:") || api.getWsdlUrl()
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: line length 120 exceeded?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java (hunk 1 above)
Comment (tharikaGitHub) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java]: Fixed

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java (hunk 2 above)
Comment (tharikaGitHub) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: Line length is 120

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java (hunk 3 above)
Comment (tharikaGitHub) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: Line length is 119

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 4]
```diff
@@ -1733,6 +1733,17 @@ public static ApplicationManagementServiceClient getApplicationManagementService
...
+    /**
+     * Method used to create the file name of the wsdl to be stored in the registry
+     *
+     * @param provider   Name of the provider of the API
+     * @param apiName    Name of the API
+     * @param apiVersion API Version
+     * @return WSDL file name
+     */
+    public static String createWsdlFileName(String provider, String apiName, String apiVersion) {
+        return provider + "--" + apiName + apiVersion + ".wsdl";
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: shouldn't we just remove the provider from here?

Comment (tharikaGitHub) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: The registry will store the wsdl file with the provider name appended. Hence it cannot be removed from this method.

**Pull request merged**